test/runner/tob.ts

```diff
@@ -0,0 +1,144 @@
+import { printSourceFile } from '../../src/compiler/printer';
+import { parseScript, parseModule } from '../../src/kataw';
+import { promiseToReadFile, promiseToWriteFile, Constants, deepEqual } from './utils';
+
+// testing object
+export interface Tob {
+  filename: string;
+  content?: string;
+  input: string;
+  parserOptions: any;
+  printerOptions: any;
+  cst?: any;
+  $cst?: any;
+  printed?: any;
+  $printed?: any;
+  diagnostics?: any;
+  $diagnostics?: any;
+  isMatched?: boolean;
+}
+
+export function updateTob(tob: any, updateItem: any): void {
+  (updateItem === 'all' || updateItem === 'parser') && (tob.cst = tob.$cst);
+  (updateItem === 'all' || updateItem === 'parser') && (tob.printed = tob.$printed);
+  (updateItem === 'all' || updateItem === 'parser') && (tob.diagnostics = tob.$diagnostics);
+  let outputIndex = tob.content.indexOf(Constants.Output);
+  const output = outputBlock(tob);
+  tob.content = tob.content.slice(0, outputIndex) + output;
```

Comment:
Slice here will cause - on windows - that the entire .md file content to be part of "obj.content" and not only the AST that you should update. That again get written to file and are causing the mentioned issue.

Reply:
@aladdin-add `list.forEach((obj: any) => (obj.newoutput = {}));` existed before in `cli.ts` but adding it back doesn't solve  the issue

Reply:
the `.newoutput` was still there ( in main branch).